src/GameState.cpp

```diff
@@ -1267,38 +1267,38 @@ void GameState::ResetStageStatistics()
 	m_iStageSeed = rand();
 }
 
-void GameState::UpdateSongPosition( float fPositionSeconds, const TimingData &timing, const RageTimer &timestamp )
+void GameState::UpdateSongPosition(double fPositionSeconds, const TimingData& timing, const RageTimer& timestamp)
 {
 	/* It's not uncommon to get a lot of duplicated positions from the sound
 	 * driver, like so: 13.120953,13.130975,13.130975,13.130975,13.140998,...
 	 * This causes visual stuttering of the arrows. To compensate, keep a
 	 * RageTimer since the last change. */
-	if(fPositionSeconds == m_LastPositionSeconds && !m_paused)
+	if (fPositionSeconds == m_LastPositionSeconds && !m_paused)
 		fPositionSeconds += m_LastPositionTimer.Ago();
 	else
 	{
 		//LOG->Info("Time difference: %+f",
-		//	m_LastPositionTimer.Ago() - (fPositionSeconds - m_LastPositionSeconds)
+		//  m_LastPositionTimer.Ago() - (fPositionSeconds - m_LastPositionSeconds)
 		//);
 		m_LastPositionTimer.Touch();
 		m_LastPositionSeconds = fPositionSeconds;
 	}
 
-	m_Position.UpdateSongPosition( fPositionSeconds, timing, timestamp );
+	m_Position.UpdateSongPosition(fPositionSeconds, timing, timestamp);
 
-	FOREACH_EnabledPlayer( pn )
+	FOREACH_EnabledPlayer(pn)
 	{
-		if( m_pCurSteps[pn] )
+		if (m_pCurSteps[pn])
 		{
-			float fAdditionalVisualDelay = m_pPlayerState[pn]->m_PlayerOptions.GetPreferred().m_fVisualDelay;
-			m_pPlayerState[pn]->m_Position.UpdateSongPosition( fPositionSeconds, *m_pCurSteps[pn]->GetTimingData(), timestamp, fAdditionalVisualDelay );
-			Actor::SetPlayerBGMBeat( pn, m_pPlayerState[pn]->m_Position.m_fSongBeatVisible, m_pPlayerState[pn]->m_Position.m_fSongBeatNoOffset );
+			double fAdditionalVisualDelay = m_pPlayerState[pn]->m_PlayerOptions.GetPreferred().m_fVisualDelay;
```

Comment:
The value is technically a float. If you want it to function as a double, you'll probably want to update the function definition for m_Position.

Reply:
Hmm, I'm going to hold off on that because changing stuff in `m_Position` will cause data type errors to crop up elsewhere. Reverting.